synapse/storage/database.py

```diff
@@ -305,15 +307,11 @@ def execute_values(self, sql: str, *args: Any, fetch: bool = True) -> List[Tuple
         from psycopg2.extras import execute_values
 
         return self._do_execute(
-            # Type ignore: mypy is unhappy because if `x` is a 5-tuple, then there will
-            # be two values for `fetch`: one given positionally, and another given
-            # as a keyword argument. We might be able to fix this by
-            # - propagating the signature of psycopg2.extras.execute_values to this
-            #   function, or
-            # - changing `*args: Any` to `values: T` for some appropriate T.
-            lambda *x: execute_values(self.txn, *x, fetch=fetch),  # type: ignore[misc]
+            lambda the_sql, the_values: execute_values(
+                self.txn, the_sql, the_values, fetch=fetch
```

Comment:
(This code was pre-existing, but...)

`the_sql` doesn't change for each iterate, correct? So really no reason to pass it in for each one?

Reply:
Sorry, what do you mean?

`the_sql` is given by `_do_execute`.

`the_values` are just a pass-through of `values` and could have been part of the lambda's closure, but in analogue to `execute_batch` (and indeed the way this code was written before), it's passed through. If we want to change this I suppose I have no objections, though I wonder if the idea was that `_do_execute` could have transformed the args if necessary...?

Reply:
Errr, yes...that didn't make sense.

My confusion is -- why are we passing in `fetch` directly into the lambda, but we pass in `the_sql` and `the_values` as arguments to the lambda?

Reply:
Ah, I did not realize that `_do_execute` modified the passed in `sql`!

Reply:
I've changed this around a bit, is that better to you?

Reply:
I do find that clearer, yes!